smqtk_descriptors/impls/descriptor_generator/pytorch.py

```diff
@@ -0,0 +1,538 @@
+import abc
+import copy
+import itertools
+import logging
+
+import numpy as np
+
+from typing import (
+    Any, Callable, Dict, Iterable, Iterator, Mapping,
+    Optional, Sequence, Set, Type, TypeVar, Union
+)
+
+from smqtk_dataprovider import DataElement
+from smqtk_descriptors import DescriptorGenerator
+from smqtk_image_io import ImageReader
+from smqtk_core.configuration import (
+    from_config_dict,
+    make_default_config,
+    to_config_dict,
+)
+
+from smqtk_descriptors.utils.parallel import parallel_map
+from smqtk_descriptors.utils.pytorch_utils import load_state_dict
+
+
+LOG = logging.getLogger(__name__)
+
+try:
+    import torch  # type: ignore
+    from torch.utils.data import Dataset  # type: ignore
+    import torchvision.models  # type: ignore
+    import torchvision.transforms  # type: ignore
+    from torch.nn import functional as F  # type: ignore
+except ModuleNotFoundError as ex:
+    LOG.warning(f"Failed to import torch module: {ex}")
+    torch = None  # type: ignore
+
+__all__ = [
+    "TorchModuleDescriptorGenerator",
+    "Resnet50SequentialTorchDescriptorGenerator",
+    "AlignedReIDResNet50TorchDescriptorGenerator"
+]
+T = TypeVar("T", bound="TorchModuleDescriptorGenerator")
+
+
+def normalize_vectors(v: np.ndarray,
+                      mode: Optional[Union[int, float, str]] = None) -> np.ndarray:
+    """
+    Array/Matrix normalization along max dimension (i.e. a=0 for 1D array, a=1
+    for 2D array, etc.).
+
+    :param np.ndarray v:
+        Vector to normalize.
+    :param None|int|float|str mode:
+        ``numpy.linalg.norm`` order parameter.
+
+    :return: Normalize version of the input array ``v``.
+    :rtype: np.ndarray
+    """
+    if mode is not None:
+        n = np.linalg.norm(v, mode, v.ndim - 1, keepdims=True)
+        # replace 0's with 1's, preventing div-by-zero
+        n[n == 0.] = 1.
+        return v / n
+    # When normalization off
+    return v
+
+
+try:
+    class ImgMatDataset (Dataset):
+
+        def __init__(self,
+                     img_mat_list: Sequence[np.ndarray],
+                     transform: Callable[[Iterable[np.ndarray]], "torch.Tensor"]) -> None:
```

Comment:
We can alternatively  split up long parameter lines like:
```suggestion
        def __init__(
            self,
            img_mat_list: Sequence[np.ndarray],
            transform: Callable[[Iterable[np.ndarray]], "torch.Tensor"]
        ) -> None:
```